modules/__tests__/LocationUtils-test.js

```diff
@@ -146,6 +152,22 @@ describe('LocationUtils', () => {
           expect(location.query).toBe(null)
         })
       })
+
+      describe('key', () => {
+        it('is the provided key', () => {
+          const descriptor = {
+            key: 42
+          }
+          const location = createRouterLocation(descriptor, parse, stringify)
+          expect(location.key).toBe(descriptor.key)
+        })
+
+        it('is undefined when there is no key', () => {
+          const descriptor = {}
+          const location = createRouterLocation(descriptor, parse, stringify)
+          expect(location.key).toBe(undefined)
```

Comment:
I suggest using `expect(location).toExcludeKey('key')` here.
While `expect(location.key).toBe(undefined)` will succeed for `location = {key: undefined}`, `expect(location.key).toBe(undefined)` will not.
See #4109.